src/rtc.rs

```diff
@@ -0,0 +1,136 @@
+use stm32::{RTC};
+
+use crate::rcc::Clocks;
+
+
+/*
+    Configuring RTC registers requires the following process:
+    poll RTOFF, make sure it is 1
+    set CNF to enter config mode
+    write to the registers
+    clear CNF
+    poll RTOFF to make sure it's back to 1
+    The frequency of the rtc is calculated by
+    f_rtcclk / (PRL[19:0] + 1)
+    f_rtcclk can probably be configured to be the main clock. Then we can use
+    rcc to figure out what that frequency is
+    both vbat and vdd might have to be connected for the clock to keep counting
+*/
+
+/**
+  Interface to the real time clock. This struct can not be instanciated directly,
+  instead, it is created by the `BackupDomain`. This ensures that the rtc registers
+  are writeable and that the clock has been configured correctly.
+*/
+pub struct Rtc {
+    regs: RTC
+}
+
+
+impl Rtc {
+    /**
+      Initialises the RTC, this should only be called if access to the backup
+      domain has been enabled
+    */
+    pub(crate) fn rtc(regs: RTC, _clocks: &Clocks) -> Self {
+        // Set the prescaler to make it count up once every second
+        // The manual on page 490 says that the prescaler value for this should be 7fffh
+        regs.prll.write(|w| unsafe{w.bits(0x7fff)});
+        regs.prlh.write(|w| unsafe{w.bits(0)});
+
+        Rtc {
+            regs
+        }
+    }
+
+    /// Starts an alarm which will trigger the RTCALARM interrupt in `time_seconds`
+    pub fn set_alarm(&mut self, time_seconds: u32) {
+        // Reset counter
+        self.perform_write(|s| {
+            s.regs.cnth.write(|w| unsafe{w.bits(0)});
+        });
+        self.perform_write(|s| {
+            s.regs.cntl.write(|w| unsafe{w.bits(0)});
+        });
```

Comment:
This in contrary to my usecase: I need to have cnt to be epoch for my alarm clock. For me, set alarm ask for a date, but must not change cnt.

Reply:
Good point, that's probably a more general use case. I'll make it not reset the count and while i'm at it I'll add a `set_cnt` function